File: test/resources/HelloWorld/pipelines/customers_orders/code/job/graph/By_CustomerId.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from job.config.ConfigStore import *
from job.udfs.UDFs import *

def By_CustomerId(spark: SparkSession, in0: DataFrame, in1: DataFrame, ) -> DataFrame:
    return in0\
        .alias("in0")\
        .join(in1.alias("in1"), (col("in0.customer_id") == col("in1.customer_id")), "inner")\
        .select(col("in1.account_open_date").alias("account_open_date"), col("in0.order_id").alias("order_id"), col("in0.customer_id").alias("customer_id"), col("in0.amount").alias("amount"), col("in1.first_name").alias("first_name"), col("in1.last_name").alias("last_name"))

File: test/resources/HelloProphecy/prophecy_scala/pipelines/silver_customers_orders/code/src/main/scala/io/prophecy/pipelines/silver_customers_orders/graph/ByCustomerId.scala
package io.prophecy.pipelines.silver_customers_orders.graph

import io.prophecy.libs._
import io.prophecy.pipelines.silver_customers_orders.udfs.PipelineInitCode._
import io.prophecy.pipelines.silver_customers_orders.udfs.UDFs._
import io.prophecy.pipelines.silver_customers_orders.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object ByCustomerId {

  def apply(context: Context, in0: DataFrame, in1: DataFrame): DataFrame =
    in0
      .as("in0")
      .join(in1.as("in1"),
            col("in0.customer_id") === col("in1.customer_id"),
            "inner"
      )
      .select(
        col("in0.order_id").as("order_id"),
        col("in0.order_date").as("order_date"),
        col("in0.amount").as("amount"),
        col("in0.customer_id").as("customer_id"),
        col("in1.account_open_date").as("account_open_date"),
        col("in1.first_name").as("first_name"),
        col("in1.last_name").as("last_name"),
        col("in1.zipcode").as("zipcode"),
        col("in1.account_flags").as("account_flags")
      )

}
